### ISSUE TITLE
I like to see es6 source map

### ISSUE THREAD

**Issue by khomyakov42** [ember-cli/ember-cli#4528]
want to see in the source maps es6 syntax
```
import Ember from 'ember';
import Resolver from 'ember/resolver';
import loadInitializers from 'ember/load-initializers';
import config from './config/environment';

var App;

Ember.MODEL_FACTORY_INJECTIONS = true;

App = Ember.Application.extend({
  modulePrefix: config.modulePrefix,
  podModulePrefix: config.podModulePrefix,
  Resolver: Resolver
});

loadInitializers(App, config.modulePrefix);

export default App;
```
, but I see the compiled code
```

define('simple-ember/app', ['exports', 'ember', 'ember/resolver', 'ember/load-initializers', 'simple-ember/config/environment'], function (exports, Ember, Resolver, loadInitializers, config) {

  'use strict';

  var App;

  Ember['default'].MODEL_FACTORY_INJECTIONS = true;

  App = Ember['default'].Application.extend({
    modulePrefix: config['default'].modulePrefix,
    podModulePrefix: config['default'].podModulePrefix,
    Resolver: Resolver['default']
  });

  loadInitializers['default'](App, config['default'].modulePrefix);

  exports['default'] = App;

});
```
how can I get this behavior by ember-cli?

**Comment by stefanpenner** [comment]
Soon. Once we make babel also do module transpiration. If all works as planned it should be in the next 2 weeks

**Comment by glasscube42** [comment]
This is fixed?

**Comment by stefanpenner** [comment]
@glasscube42 nope, issue is still open.

https://github.com/ember-cli/ember-cli/pull/4764 gets us closer

**Comment by peta** [comment]
+1 --- Ran against this issue earlier today and spent several hours on it before becoming aware that this is a (very annoying) bug. Still persists in altest ember-cli.

**Comment by stefanpenner** [comment]
So we have it working on master. But the cost of sourcemap concat becomes prohibitive it is unlikely to be enabled by default until the source map library is improved. Or maybe optionally enable chrome only (until others implement) source map sections

**Comment by peta** [comment]
@stefanpenner Great news. :) I just npm installed the latest ember-cli right from the Github repository -- but I didn't manage to ge tit right. Could you tell me what's wrong with my config file: https://gist.github.com/anonymous/1d67a7fba0b1ab14f8ff

**Comment by stefanpenner** [comment]
> @stefanpenner Great news. :)

we have explicitly disabled it: https://github.com/ember-cli/ember-cli/blob/master/lib/broccoli/ember-app.js#L209-L211

**Comment by stefanpenner** [comment]
closing, basically we have support for this, but the source-map module is to slow for realistic use.

We have been working on https://github.com/krisselden/fast-source-map hopefully sometime soon we will have time to land this

**Comment by stefanpenner** [comment]
Just an update, we have been focused on other areas. But plan to revive the fast-source-map effort.

As before, we do technically support this, but the current source-map libraries in existence, can't handle reasonably sized source-map concatenations in a performant way. The above effort will hopefully help.

We also have plans to allow for http2 or source-map sections work, but that will land after the fast-source-map work, as that will work for all supported platforms.

**Comment by JonET** [comment]
:+1:

**Comment by topaxi** [comment]
:+1: I was trying to build an istanbul plugin, but as it tries to remap with sourcemaps, which don't map to the original files, it fails with an hard error.

**Comment by iivvoo** [comment]
Any updates on this? I would love to see this working and have a less horrible debugging experience.

Additonally, vscode has pretty good remote debugging support (it can connect to a running chrome) but it completely fails with Ember because the source maps don't map. It would be awesome if you could debug your original code directly this way.

**Comment by nathanhammond** [comment]
This area continues to change and we're paying attention to it:
https://github.com/emberjs/core-notes/pull/108/files#diff-4653b59711e4e3ef1159f793dd2c597cR29

I'm closing this issue as our changes are going to be incremental. Note that we are of multiple opinions about the value of source maps on the core team. For example, my ideal default would be to add in sourcemaps only to map uglified code into non-uglified code. Other transformations (e.g. Babel) can result in introducing bugs themselves and debugging at a layer up from that transform makes it difficult to catch bugs in your running code.

The way out of this mess requires multiple sourcemap support but that's not supported in any tools at this time.

**Comment by Kerrick** [comment]
@nathanhammond Is there another issue (or meta-issue since the changes will be incremental) that we can track for this?

**Comment by dwickern** [comment]
@Kerrick you can enable sourcemaps using this option: https://github.com/babel/ember-cli-babel#enabling-source-maps

**Comment by Kerrick** [comment]
@dwickern Yes, but they still don't show the original ES6+ code. I see sourcemaps with individual files for each component, for example, but I don't see arrow functions, `import` statements, ES6-style method declarations, etc. That's what I took this issue to be about -- the accuracy of the sourcemaps, not how to turn them on.

**Comment by dwickern** [comment]
@Kerrick If that option doesn't work, you may need to upgrade ember-cli.

You need at least ember-cli v2.12.0-beta.1 for #6458
If you're on a more recent ember-cli, you need at least ember-cli-babel v6.4.2 for babel/ember-cli-babel#158

Here's what you see with a new ember app:
![screen shot 2017-07-31 at 11 57 02 am](https://user-images.githubusercontent.com/752885/28793322-7d17974c-75e7-11e7-92a0-80ff592a7686.png)

Here's what you should see after enabling sourcemaps:
![screen shot 2017-07-31 at 11 57 40 am](https://user-images.githubusercontent.com/752885/28793329-7f95b396-75e7-11e7-8bb2-b5924a807e47.png)

**Comment by Kerrick** [comment]
Ah, that's likely to be my issue then -- our version of `ember-cli-babel` is pinned to 6.4.1. Thanks!

**Comment by dwickern** [comment]
FYI, source maps are broken again on latest ember-cli ef4/fast-sourcemap-concat#39

**Comment by pixelpax** [comment]
I'm still not seeing the first pane which @dwickern posted, rather than the second with source-maps enabled. 

I'm using ember-cli-babel@6.11.0 and ember-cli@2.15.1

**Comment by dwickern** [comment]
@pixelpax We're currently blocked on ef4/fast-sourcemap-concat#39

You can work around it using my fork by adding this resolution to your `package.json` and running `yarn`
```
{
  "devDependencies": {
    ...
  },
  "resolutions": {
    "fast-sourcemap-concat": "dwickern/fast-sourcemap-concat#skip-validation"
  }
}
```

Then follow the instructions here https://github.com/babel/ember-cli-babel#enabling-source-maps

**Comment by pixelpax** [comment]
Hm... I tried this, but unfortunately I'm seeing the same thing. 

Would you expect this fix to also work for engine development? 

If so I could try to reproduce w/ a minimal engine setup.
